.githooks/pre-push

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
```

Comment:
I reviewed the GH issue and only the comments at the top of this file, but not the code in place.

Doing a diff of `diff .githooks/pre-push .githooks/pre-commit`, seems like a lot of logic is shared, so I was wondering what your thoughts are on extracting the common logic/helpers into a separate bash file? The goal is to  have it potentially be reusable in our scripts/hooks in the future.

I realize this is a fair amount of extra work, so don't feel strongly about it.

Lmk what you think

Reply:
Haha - indeed! I actually had to restrain myself from attempting a refactor as I was unsure about the portability of something like `declare -g ...` which was the first thing I would've reached for in a first pass. To avoid that, I think one would have to think more comprehensively about how the script could be broken down into more functional parts that are compatible with bash's calling conventions, and no doubt, consume some more reference material. As it's currently written, each loop modify multiple arrays declared in a common scope.

Reply:
❤️ this reasoning